Userland/Utilities/man.cpp

```diff
@@ -59,53 +60,68 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     TRY(Core::System::unveil("/bin", "x"));
     TRY(Core::System::unveil(nullptr, nullptr));
 
-    char const* section = nullptr;
+    struct Section {
+        char const* number;
+        char const* category;
+    };
+    Section manpage_data = { nullptr, nullptr };
     char const* name = nullptr;
     char const* pager = nullptr;
 
     Core::ArgsParser args_parser;
     args_parser.set_general_help("Read manual pages. Try 'man man' to get started.");
-    args_parser.add_positional_argument(section, "Section of the man page", "section", Core::ArgsParser::Required::No);
+    args_parser.add_positional_argument(manpage_data.number, "Section of the man page", "section", Core::ArgsParser::Required::No);
     args_parser.add_positional_argument(name, "Name of the man page", "name");
     args_parser.add_option(pager, "Pager to pipe the man page to", "pager", 'P', "pager");
     args_parser.parse(arguments);
 
-    auto make_path = [name](char const* section) {
-        return String::formatted("/usr/share/man/man{}/{}.md", section, name);
+    auto make_path = [name](char const* number) {
+        return String::formatted("/usr/share/man/man{}/{}.md", number, name);
+    };
+    Array sections = {
+        Section { "1", "User Programs" },
+        Section { "2", "System Calls" },
+        Section { "3", "Library Functions" },
+        Section { "4", "Special Files" },
+        Section { "5", "File Formats" },
+        Section { "6", "Games" },
+        Section { "7", "Miscellanea" },
+        Section { "8", "Sysadmin Tools" }
     };
-    if (!section) {
-        char const* sections[] = {
-            "1",
-            "2",
-            "3",
-            "4",
-            "5",
-            "6",
-            "7",
-            "8"
-        };
-        for (auto s : sections) {
-            String path = make_path(s);
+    if (manpage_data.number) {
+        String path = make_path(manpage_data.number);
+        if (access(path.characters(), R_OK) == 0) {
+            for (Section section : sections) {
+                if (*section.number == *manpage_data.number) {
+                    manpage_data.category = section.category;
+                    break;
+                }
+            }
+        }
```

Comment:
Is this bit really needed? I doesn't look to me like this `access(path)` check was done here before when the manpage number was provided (and you can see it's done again below). 

I think this whole bit could be 
```c++
auto manpage_number = TRY(manpage_data.number.to_int());
manpage_data.category = sections[manpage_number - 1].category;
```